core/session.go

```diff
@@ -0,0 +1,154 @@
+package core
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/dicedb/dice/config"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	DefaultUserName = "default"
+	AuthCmd         = "AUTH"
+
+	SessionStatusPending SessionStatusT = SessionStatusT(0)
+	SessionStatusActive  SessionStatusT = SessionStatusT(1)
+	SessionStatusExpired SessionStatusT = SessionStatusT(2)
+)
+
+var (
+	UserStore *Users = NewUsers()
+)
+
+type (
+	SessionStatusT uint8
+	Session        struct {
+		ID   uint64
+		User *User
+
+		CreatedAt      time.Time
+		LastAccessedAt time.Time
+
+		Status SessionStatusT
+	}
+
+	Users struct {
+		store  map[string]*User
+		stLock *sync.RWMutex
+	}
+
+	User struct {
+		Username          string
+		Passwords         []string
```

Comment:
Why multiple password for one user?

Reply:
Redis allows multiple passwords per single error. It compares all the passwords and returns authentication success if one of the set passwords match.